### ISSUE TITLE
Eraser support

### ISSUE THREAD

**Issue by personalizedrefrigerator** [personalizedrefrigerator/js-draw#52]
Currently, the eraser button on physical styluses does nothing.

At present, this is because Chromium seems to not report stylus events when the stylus is down (just when the stylus is above the screen), or a bug in how `js-draw` is registering pointer event listeners.

Even if we can only listen for button events when the pointer is above the screen, pressing the button could be used to, for example, switch tools.

**Comment by Wladefant** [comment]
If it is only possible to listen for button events when the pointer is above the screen, it is still perfect for using the eraser. Samsung Notes, for example, activates the eraser while holding the button and switches back to the normal pen when lifting the finger off. 

Or was that what you were talking about, that that is not possible?

**Comment by personalizedrefrigerator** [comment]
At least with my tablet, this is what happens when I try to draw with the eraser button down:
1. Pushing the eraser button while the pen is above the screen: Pointer move events now have `buttons: 1`,
2. Touching the pen to the screen: Pointer events are no longer received
3. Lifting the pen from the screen: Pointer events are received again, with `buttons: 1`

This can be seen by checking the "live update" checkbox on the [input logging](https://js-draw.web.app/debugging/stroke-logging/) page and pushing the eraser button.

<details><summary>Input log</summary>

[[uploading...]]

</details>

**Comment by Wladefant** [comment]
I tried it. While connected to air actions it definitely registers the button press. Dont know how to test whether it works with the button pressed for a long time. There is no reaction for that

**Comment by personalizedrefrigerator** [comment]
Samsung Internet **does** seem to send pointer events when drawing with the eraser button pressed. However, these pointer events have `buttons: 1`. According to the [Pointer Events spec](https://w3c.github.io/pointerevents/#the-buttons-property), however, this means "Left Mouse, Touch contact, Pen contact".

**Comment by Wladefant** [comment]
Yes i see that. I didnt get any event when eraser buttons is pressed. Strange. It should be 32

**Comment by yajo** [comment]
FWIW now that this lib is included in Joplin for handwritten notes, this would improve a lot the note-taking experience in Samsung tablets with S-Pen.

**Comment by personalizedrefrigerator** [comment]
[This upstream Chromium bug report seems to be the same (or a similar) issue referenced here.](https://bugs.chromium.org/p/chromium/issues/detail?id=1371798&q=stylus&can=2)

**Comment by Wladefant** [comment]
the chinese app touchnotes supports erasing with a samsung pen. No other pen events are working, but the pressing and holding event does work, so there is defeinitely a work around for there

**Comment by Wladefant** [comment]
Did you try to fix it in May. What should I try?

**Comment by personalizedrefrigerator** [comment]
> the chinese app touchnotes supports erasing with a samsung pen. No other pen events are working, but the pressing and holding event does work, so there is defeinitely a work around for there

In Joplin, `js-draw` runs within a [WebView](https://developer.android.com/reference/android/webkit/WebView), which (on most systems) is based on Chromium. From a brief web search, it looks like [touchnotes](https://play.google.com/store/apps/details?id=com.newskyer.draw&hl=en_US) is a native app. Based on this and [the upstream bug report](https://issues.chromium.org/issues/40241490), I suspect that the issue is related to Chromium and not the native Android APIs.

Possible next steps:
- The [upstream bug report](https://issues.chromium.org/issues/40241490) states that the issue doesn't happen on Windows — confirm this.
- **(Difficult to test & maintain)**: Look into capturing native pointer events on Android and forwarding them to the image editor running in a Web View.
- **(Difficult)**: Create a custom build of Chromium and try to debug and/or fix the issue.
